packages/devtools_app/lib/src/standalone_ui/vs_code/debug_sessions.dart

```diff
@@ -154,9 +154,17 @@ class _DevToolsMenuState extends State<_DevToolsMenu> {
   void _initExtensions() {
     final sessionRootPath = widget.session.projectRootPath;
     if (sessionRootPath != null) {
-      final fileUri = Uri.file(sessionRootPath);
-      _extensionServiceForSession =
-          ExtensionService(fixedAppRootPath: fileUri.toString());
+      // This file path might be a Windows path but because this code runs in
+      // the web, Uri.file() will not handle it correctly.
+      //
+      // Since all paths are absolute, assume that if the path does not start
+      // with '/' that it's Windows, and parse with Windows semantics.
+      final fileUri = Uri.file(
+        sessionRootPath,
+        windows: !sessionRootPath.startsWith('/'),
+      );
```

Comment:
This is the actual fix.. There are a few other ways we could do this:

```dart
Uri.file(x, windows: !x.startsWith('/')) // assume if doesn't start with / it's Windows
Uri.file(x, windows: userAgent.contains('win')) // assume Windows if browser is Windows
Uri.file(x, windows: x.startsWith('\[a-zA-Z](?::|%3a|%3A)]')) // assume Windows path starts with drive letter and then colon (or encoded colon)
```

In hindsight, it would've been better if the tooling API was passing file URIs in the protocol.. that's more of a breaking change though. We could add it as a new field and use that in preference with the goal of dropping the "path" version (far into the) future.. lmk if you think that's worthwhile.

Reply:
> Uri.file(x, windows: userAgent.contains('win')) // assume Windows if browser is Windows

I actually like this one. I would worry that somehow we'd be missing a leading slash on a non-windows uri and then parse it incorrectly as windows. Another option is to OR these checks if we don't trust the robustness of any individually.

> In hindsight, it would've been better if the tooling API was passing file URIs in the protocol

What about making this the default when we switch from using postMessage to using the DTD to communicate between VS code and DevTools?

Reply:
> > ```
> > Uri.file(x, windows: userAgent.contains('win')) // assume Windows if browser is Windows
> > ```
> 
> I actually like this one. 

My concern doing only this is that "win" is quite short and who knows what other things might be in peoples user agents. Maybe for now we should do `userAgent.contains('win') && !sessionRootPath.startsWith('/')`. I'll verify the user agent when embedded in VS Code and push that unless you'd prefer something else.

> What about making this the default when we switch from using postMessage to using the DTD to communicate between VS code and DevTools?

Sounds like a plan to me. If we can standardise on URIs everywhere (and ensuring DevTools in the browser never gets paths so it doesn't need to use `Uri.file()` or `Uri.toFilePath()`) it should avoid a number of gotchas here.

And ofc, we should work on getting Windows bots running. I suspect a lot of the shell scripts can be converted to devtools_tool. but if there are bits that need to be Windows shell scripts (to bootstrap setting up `devtools_tool`) I'm happy to work on them.